.github/workflows/main.yml

```diff
@@ -43,18 +121,26 @@ jobs:
           # We then uninstall it otherwise tests fails due to pytest hook being
           # registered twice.
           pip uninstall -y pytest-pyodide
-          npm install node-fetch@2
-      - name: Download Pyodide
-        shell: bash -l {0}
-        run: |
-          wget https://github.com/pyodide/pyodide/releases/download/0.21.0a3/pyodide-build-0.21.0a3.tar.bz2
-          tar xjf pyodide-build-0.21.0a3.tar.bz2
-          mv pyodide dist/
+          which npm && npm install node-fetch@2
+
+      - name: Get Pyodide from cache
+        uses: actions/cache@v3
+        with:
+          path: dist
+          key: ${{ runner.os }}-pyodide-$PYODIDE_VERSION-${{ hashFiles('.github/**/*.yml') }}
+
       - name: Run tests
         shell: bash -l {0}
         run: |
-          pytest --cov=pytest_pyodide -v --dist-dir=./dist/
+          pytest -v \
+            --cov=pytest_pyodide \
+            --dist-dir=./dist/ \
+            --runner=${{ matrix.test-config.runner }} \
+            -k "${{ matrix.test-config.browser }}"
+
       - uses: codecov/codecov-action@v3
-        if: ${{ github.event.repo.name == 'pyodide/pytest-pyodide' }}
+        if: ${{ github.event.repo.name == 'pyodide/pytest-pyodide' &&
+                matrix.test-config.runner == 'selenium' &&
+                matrix.test-config.browser == 'chrome' }}
```

Comment:
Normally we should upload it for each job, and it would just aggregate them.

Reply:
BTW I don't get why we have no CI status in codecov while it is being correctly uploaded. Looks like it's still not enabled somewhere (though I did check all the expected config places).

Reply:
> Normally we should upload it for each job, and it would just aggregate them.

Thanks! I didn't know that.

> I don't get why we have no CI status in codecov while it is being correctly uploaded.

It was not uploading codecov for pull requests, it seems like the [config I added](https://github.com/pyodide/pytest-pyodide/pull/4/files#diff-7829468e86c1cc5d5133195b5cb48e1ff6c75e3e9203777f6b2e379d9e4882b3R58) in #4 was wrong.